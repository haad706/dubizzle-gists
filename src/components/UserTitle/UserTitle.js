import styled from "styled-components";
import PropTypes from "prop-types";

export const UserTitle = ({ imageSrc, name }) => {
  return (
    <Wrapper>
      <IconWrapper src={imageSrc} />
      <p>{name}</p>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: inline-flex;
  gap: 0.5rem;
  justify-content: center;
  align-items: center;
  font-weight: 500;
`;

const IconWrapper = styled.img.attrs((props) => ({
  src: props.src,
}))`
  width: 50px;
  height: 50px;
  border-radius: 50%;
`;

UserTitle.propTypes = {
  name: PropTypes.string.isRequired,
  imageSrc: PropTypes.string.isRequired,
};

export default UserTitle;
