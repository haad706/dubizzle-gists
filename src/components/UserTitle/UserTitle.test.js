import { render } from "@testing-library/react";
import UserTitle from "./UserTitle";

describe("UserTitle", () => {
  it("should render successfully", () => {
    const { baseElement } = render(
      <UserTitle
        imageSrc="https://avatars.githubusercontent.com/u/127468175?v=4"
        name="dubizzle"
      />
    );
    expect(baseElement).toBeTruthy();
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <UserTitle
        imageSrc="https://avatars.githubusercontent.com/u/127468175?v=4"
        name="dubizzle"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
