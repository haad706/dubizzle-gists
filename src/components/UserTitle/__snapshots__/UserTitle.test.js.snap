// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTitle should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bgqQcB jMWrcl"
  >
    <img
      class="sc-gTRrQi iFVbbH"
      src="https://avatars.githubusercontent.com/u/127468175?v=4"
    />
    <p>
      dubizzle
    </p>
  </div>
</DocumentFragment>
`;
